src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonNamingConventionClassExtendsClassCtoCheck.java

```diff
@@ -0,0 +1,24 @@
+package com.devonfw.ide.sonarqube.common.impl.check;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+
+/**
+ * @author vhacimuf
+ *
+ */
+
+@Rule(key = "Devon4j:S2", name = "Devon Naming Convention Check", //
+    description = "Verify that Classes extending AbstractCto shall end with Cto.", //
+    priority = Priority.CRITICAL, tags = { "bug" })
+
+public class DevonNamingConventionClassExtendsClassCtoCheck extends DevonNamingConventionClassExtendsClassCheck {
+
+  @Override
+  public void init() {
+
+    this.extendedSuperClass = "AbstractCto";
+    this.classSuffixRegEx = "Cto$";
```

Comment:
such initialization should IMHO go to the constructor. I would even recommend to make these fields final to prevent desasters.

Reply:
Otherwise declare them as getters that are abstract in the parent class.

Reply:
Thank you so much for the feedback. That's a good point. In my last commit, I revised the architecture. I'm using the superclass constructor to initialize the final attributes